index.js

```diff
@@ -447,16 +449,23 @@ function getEncodingExtension (encoding) {
 }
 
 function getRedirectUrl (url) {
-  if (url.startsWith('//') || url.startsWith('/\\')) {
-    // malicous redirect
-    return '/'
+  let i = 0
+  // we detech how many slash before a valid path
+  for (i; i < url.length; i++) {
+    if (url[i] !== '/' && url[i] !== '\\') break
   }
+  // turns all leading / or \ into a single /
+  url = '/' + url.substr(i)
   try {
     const parsed = new URL(url, 'http://localhost.com/')
     return parsed.pathname + (parsed.pathname[parsed.pathname.length - 1] !== '/' ? '/' : '') + (parsed.search || '')
   } catch (error) {
+    // the try-catch here is actually unreachable, but we keep it for safety and prevent DoS attack
```

Comment:
In here, the `catch` branch is actually unreachable after we convert all leading slash to a single `/`. But, it may be changed in future and `URL` will be crashed on some weird case.

So, I will keep the `try {} catch {}` here and ignore the coverage.

Why it is non reachable?
When we specify the `host` for in `URL`, the `path` can be either a `full url` and `path`. We stripe the leading slash and it can only be `path` or `schema://something` in this case. Any invalid encoding components is not checked by `URL`. Thus, it is not able to crash anymore.

Reply:
This code was reachable for this command:

```
curl --path-as-is "http://localhost:3000//^/.."
```